azure/services/bastionhosts/bastionhosts.go

```diff
@@ -56,28 +58,33 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "bastionhosts.Service.Reconcile")
 	defer done()
 
-	azureBastionSpec := s.Scope.BastionSpec().AzureBastion
-	if azureBastionSpec != nil {
-		err := s.ensureAzureBastion(ctx, *azureBastionSpec)
-		if err != nil {
-			return errors.Wrap(err, "error creating Azure Bastion")
-		}
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
+
+	var resultingErr error
+	if s.Scope.IsAzureBastionEnabled() {
+		bastionSpec := s.Scope.AzureBastionSpec()
+		_, resultingErr = s.CreateResource(ctx, bastionSpec, serviceName)
 	}
 
-	return nil
+	s.Scope.UpdatePutStatus(infrav1.BastionHostReadyCondition, serviceName, resultingErr)
+	return resultingErr
 }
 
 // Delete deletes the bastion host with the provided scope.
 func (s *Service) Delete(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "bastionhosts.Service.Delete")
 	defer done()
 
-	azureBastionSpec := s.Scope.BastionSpec().AzureBastion
-	if azureBastionSpec != nil {
-		err := s.ensureAzureBastionDeleted(ctx, *azureBastionSpec)
-		if err != nil {
-			return errors.Wrap(err, "error deleting Azure Bastion")
-		}
+	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultAzureServiceReconcileTimeout)
+	defer cancel()
+
+	var resultingErr error
+	if s.Scope.IsAzureBastionEnabled() {
+		bastionSpec := s.Scope.AzureBastionSpec()
+		resultingErr = s.DeleteResource(ctx, bastionSpec, serviceName)
 	}
-	return nil
+
+	s.Scope.UpdateDeleteStatus(infrav1.BastionHostReadyCondition, serviceName, resultingErr)
```

Comment:
right now this will update the status even if there is no bastion associated with the cluster, needs to be addressed as part of #1868